pkg/controllermanager/controller/controllerregistration/extensionclusterrole/reconciler.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensionclusterrole
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+// Reconciler reconciles ClusterRoles for additional extension permissions and creates ClusterRoleBindings for binding
+// extension service accounts to such ClusterRoles.
+type Reconciler struct {
+	Client client.Client
+}
+
+// Reconcile reconciles ControllerRegistrations.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
```

Comment:
I don't see a clean-up logic for `ClusterRoleBinding`s, e.g. when the seed selector of the `ControllerRegistration` is used. Souldn't we have this?

Reply:
Can you make your example more concrete? I assume you talk about a `ControllerRegistration` being changed with a seed selector that leads to deinstallation of an extension from a certain seed. In this case, the `ServiceAccount` will be deleted which triggers a new reconciliation (and with this, a re-computation of the subjects for the `ClusterRoleBinding`).

Reply:
You're right, thanks for clarifying.